commit 7a86d66f27a8c657a3cd9fbecdf26d167c2ee92e
Author: Thomas Woerner <twoerner@redhat.com>
Date:   Wed Apr 26 15:07:18 2017 +0200

    firewall-cmd: Fix --{set,get}-{short,description} for zone
    
    The options --{set,get}-{short,description} have been used on the wrong object
    in firewall-cmd which resulted in a back trace.
    
    Fixes: RHBZ#1445238

diff --git a/src/firewall-cmd b/src/firewall-cmd
index 9988a79..1a864b3 100755
--- a/src/firewall-cmd
+++ b/src/firewall-cmd
@@ -1987,31 +1987,31 @@ if a.permanent:
         elif a.list_all_zones:
             names = fw.config().getZoneNames()
             for zone in sorted(names):
-                settings = fw.config().getZoneByName(zone)
-                cmd.print_zone_info(zone, settings.getSettings())
+                settings = fw.config().getZoneByName(zone).getSettings()
+                cmd.print_zone_info(zone, settings)
                 cmd.print_msg("")
             sys.exit(0)
 
         # set zone description
         elif a.set_description:
-            settings = fw.config().getZoneByName(zone)
+            settings = fw.config().getZoneByName(zone).getSettings()
             settings.setDescription(a.set_description)
             fw_zone.update(settings)
 
         # get zone description
         elif a.get_description:
-            settings = fw.config().getZoneByName(zone)
+            settings = fw.config().getZoneByName(zone).getSettings()
             cmd.print_and_exit(settings.getDescription())
 
         # set zone short description
         elif a.set_short:
-            settings = fw.config().getZoneByName(zone)
+            settings = fw.config().getZoneByName(zone).getSettings()
             settings.setShort(a.set_short)
             fw_zone.update(settings)
 
         # get zone short description
         elif a.get_short:
-            settings = fw.config().getZoneByName(zone)
+            settings = fw.config().getZoneByName(zone).getSettings()
             cmd.print_and_exit(settings.getShort())
 
 elif a.version:
